packages/wrangler/src/deploy/config-diffs.ts

```diff
@@ -0,0 +1,296 @@
+import { green, red } from "@cloudflare/cli/colors";
+import { Diff } from "../cloudchamber/helpers/diff";
+import type { Config, RawConfig } from "../config";
+
+/**
+ * Object representing the difference of two configuration objects.
+ */
+type ConfigDiff = {
+	/** The actual (raw) computed diff of the two objects */
+	diff: Diff;
+	/** Boolean indicating whether the difference is not present or only comprized of the addition of configuration options */
+	onlyAdditionsIfAny: boolean;
```

Comment:
Can we name this a bit more clearly? I think the intention is to mark if this diff is destructive, right? What about `nonDestructive`?

Reply:
Sure, let's go with `nonDistructive` 🙂 

it does sound a bit more abstract to me, but I am also not a fan of `onlyAdditionsIfAny` (I could not come up with anything better 😓)